packages/fee-estimation/src/estimateFees.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  gasPriceOracleABI,
+  gasPriceOracleAddress,
+} from '@eth-optimism/contracts-ts'
+import {
+  getContract,
+  createPublicClient,
+  http,
+  BlockTag,
+  Address,
+  EstimateGasParameters,
+  encodeFunctionData
+} from 'viem'
+import * as chains from 'viem/chains'
+import { PublicClient } from 'wagmi'
+
+export { encodeFunctionData }
+
+/**
+ * Bytes type representing a hex string with a 0x prefix
+ * @typedef {`0x${string}`} Bytes
+ */
+export type Bytes = `0x${string}`
+
+/**
+ * Options to query a specific block
+ */
+type BlockOptions = {
+  /**
+   * Block number to query from
+   */
+  blockNumber?: bigint
+  /**
+   * Block tag to query from
+   */
+  blockTag?: BlockTag
+}
+
+const knownChains = [chains.optimism.id, chains.goerli.id, chains.baseGoerli.id]
+
+/**
+ * ClientOptions type
+ * @typedef {Object} ClientOptions
+ * @property {keyof typeof gasPriceOracleAddress | number} chainId - Chain ID
+ * @property {string} [rpcUrl] - RPC URL. If not provided the provider will attempt to use public RPC URLs for the chain
+ * @property {chains.Chain['nativeCurrency']} [nativeCurrency] - Native currency. Defaults to ETH
+ */
+type ClientOptions =
+  // for known chains like base don't require an rpcUrl
+  | {
+    chainId: (typeof knownChains)[number]
+    rpcUrl?: string
+    nativeCurrency?: chains.Chain['nativeCurrency']
+  }
+  | {
+    chainId: number
+    rpcUrl: string
+    nativeCurrency?: chains.Chain['nativeCurrency']
+  }
+  | PublicClient
+
+/**
+ * Options for all GasPriceOracle methods
+ */
+export type GasPriceOracleOptions = BlockOptions & { client: ClientOptions }
+
+
+/**
+ * Throws an error if fetch is not defined
+ * Viem requires fetch
+ */
+const validateFetch = () => {
+  if (typeof fetch === 'undefined') {
+    throw new Error(
+      'No fetch implementation found. Please provide a fetch polyfill. This can be done in NODE by passing in NODE_OPTIONS=--experimental-fetch or by using the isomorphic-fetch npm package'
+    )
+  }
+}
+
+/**
+ * Gets L2 client
+ * @example
+ * const client = getL2Client({ chainId: 1, rpcUrl: "http://localhost:8545" });
+ */
+export const getL2Client = (options: ClientOptions): PublicClient => {
+  validateFetch()
+
+  if ('chainId' in options && options.chainId) {
+    const viemChain = Object.values(chains)?.find((chain) => chain.id === options.chainId)
+    const rpcUrls = options.rpcUrl
+      ? { default: { http: [options.rpcUrl] }, public: { http: [options.rpcUrl] } }
```

Comment:
```suggestion
      ? { default: { http: [options.rpcUrl] } }
```
Nit not sure why I am creating public here